server/datastore/mysql/policies.go

```diff
@@ -211,21 +211,23 @@ func filterNotExecuted(results map[uint]*bool) map[uint]bool {
 }
 
 func (ds *Datastore) RecordPolicyQueryExecutions(ctx context.Context, host *fleet.Host, results map[uint]*bool, updated time.Time, deferredSaveHost bool) error {
-	// Sort the results to have generated SQL queries ordered to minimize
-	// deadlocks. See https://github.com/fleetdm/fleet/issues/1146.
-	orderedIDs := make([]uint, 0, len(results))
-	for policyID := range results {
-		orderedIDs = append(orderedIDs, policyID)
-	}
-	sort.Slice(orderedIDs, func(i, j int) bool { return orderedIDs[i] < orderedIDs[j] })
-
-	// Loop through results, collecting which labels we need to insert/update
 	vals := []interface{}{}
 	bindvars := []string{}
-	for _, policyID := range orderedIDs {
-		matches := results[policyID]
-		bindvars = append(bindvars, "(?,?,?,?)")
-		vals = append(vals, updated, policyID, host.ID, matches)
+	if len(results) > 0 {
```

Comment:
Cool.
I'm curious how much this small check contributed to the overall CPU reduction. 🤔